PSAlphaFS.psm1

```diff
@@ -552,43 +576,101 @@ function Copy-LongItem
 		Else
 		{
 			$Destination_isDirectory = $true
-			if( -not ( $DirObject::Exists($Destination)))
-			{
-				$null = $DirObject::CreateDirectory($Destination)
-			} 
-			Else
+            #$DirObject::Exists seems to return false no matter what
+			#if( -not ( $DirObject::Exists($Destination)))
+			#{
+			#	$null = $DirObject::CreateDirectory($Destination)
+			#} 
+			#Else
+            #{
+			#	$Destination = $PathFSObject::Combine($Destination, $basename) 
+			#}
+            try {
```

Comment:
instead of this : 

        try {
        $null = $DirObject::CreateDirectory($Destination)
        }
        catch [Alphaleonis.Win32.Filesystem.AlreadyExistsException]{
        }

we can use this: 

            if ( -not ( $DirObject::Exists($Destination))) {

                if ($FileObject::Exists($Destination)) {
                    Write-Warning -Message ("Copy-LongItem:`tCannot create a directory at [{0}] when a file already exists with the same name" -f $Destination) 
                    return
                }
                Else {
                    try {

                        $null = $DirObject::CreateDirectory($Destination)

                    }

                    catch
                    {
                        throw $_
				
                    }            
                }

            } 

            Else {

                $Destination = $PathFSObject::Combine($Destination, $basename) 

            }